### ISSUE TITLE
yt-4.0 kdtree large memory usage

### ISSUE THREAD

**Issue by AshKelly** [yt-project/yt#1973]
### Bug report

Jared Coughlin reports:

> Hello! I'm using the sph-viz branch of yt and I'm trying to make a density projection plot from a gadget-2 snapshot. This works like a charm when I use a snapshot with 428^3 particles in a 15 Mpc box. However, when I try to do it on a snapshot with 1024^3 particles in a 40 Mpc box, my job keeps getting killed because it is trying to exceed the available system memory. I'm submitting it to a node with 256 GB of RAM, and I'm requesting all of it, which means that more than this is trying to be used. The snapshot itself is ~ 40 GB. The memory usage appears to spike when allocating for the KDTree. Before that, it appears to generate the .ewah file without issue. I was just wondering if anyone had any thoughts on why this large spike in memory might be occurring and how I might go about fixing it? If not, no worries, I can try and investigate it myself, but it's a pain to do with pdb. My gadget simulation used a maximum of ~ 173 GB of RAM, so I feel like the kdtree shouldn't be using as much memory as it's trying to. 
> Thanks, and sorry for the bother! 
> Sincerely, -Jared

this should be reproducible with large datasets,

```python
ds = GadgetDataset(file_name) 
p = yt.ProjectionPlot(ds, axis = 'z', fields = ('Gas', 'Density')).save(file_name + '.png')
```

**Actual outcome**

The code is killed due to exceeding memory usage of 256 GB of RAM.

**Expected outcome**

The code *shouldn't* be making a KDTree unless a) gather smoothing is being used, or 2) the smoothing lengths aren't found.

**Version Information**
  * Operating System: -
  * Python Version: -
  * yt version: yt-4.0
  * Other Libraries (if applicable): -

**Comment by AshKelly** [comment]
I've managed to reproduce this. The code currently loads a `KDTreeProj` and then, for some reason, the KDTree seems to eat memory and crash for large datasets.

**Comment by ngoldbaum** [comment]
Argh, that's my fault. This is related to an experiment that I was doing in the spring that we should back out now that you've done your project in a way that scales better and has acceptable performance. We really need to come up with a nice way to regenerate projections quickly, but I think my ideas about doing that using the octree will have to be shelved until we can figure out ways to scale the kdtree better.

Ash, I can do the work here since I was the one who refactored things this Spring.

**Comment by ngoldbaum** [comment]
I opened #1980.

**Comment by AshKelly** [comment]
closed by #1980
